src/Features/Lsif/Generator/Generator.cs

```diff
@@ -71,9 +71,12 @@ public static Generator CreateAndWriteCapabilitiesVertex(ILsifJsonWriter lsifJso
             return generator;
         }
 
-        public async Task GenerateForProjectAsync(Project project, GeneratorOptions options)
+        public async Task GenerateForProjectAsync(
+            Project project,
+            GeneratorOptions options,
+            CancellationToken cancellationToken)
```

Comment:
So we generally weren't worrying about cancellation at all since we don't have any way to produce a cancellation token from the start -- did you have something in mind where we might have one?

Reply:
Yup.  A totally reasonable pattern is to register to hear about ctrl-c and prompt, or use that for cancellation.  I agree this it not a big deal here as we're ok with just killing things.  But i like code with good hygiene :)